### PR TITLE
ENH: Add KBOMirrorHE and add y axis to KBOMirror status

### PR DESCRIPTION
**Pull request by nrwslac**
## Description
* Added cooling flow/pressure sensor readouts in a KBO mirror child class: KBOMirrorHE
* Added Y axis to device status output

## Motivation and Context
New hardware installed.

## How Has This Been Tested?
Ran in local pydev environment and view in a typhos screen with mirror mr2k4.

## Screenshots:
kboHE

kbo status

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Review by ZLLentz — COMMENTED** [review]
I don't have much to say, this looks simple and clean. Happy to merge once the checklist is complete.

Also, TIL that `lpm` stands for "Liters per minute". Maybe a strange unit choice for the flow rates here? But that's unrelated to this PR.

**Comment by nrwslac** [comment]
> I don't have much to say, this looks simple and clean. Happy to merge once the checklist is complete. Good job on the clear PR text too.
> 
> Also, TIL that `lpm` stands for "Liters per minute". Maybe a strange unit choice for the flow rates here? But that's unrelated to this PR.

correct, this unit was requested specifically.

**Comment by nrwslac** [comment]
One concern is kbo mirrors using this class that do not have sensors installed will show PV discon here, but that seems okay.  All KBOs will receive these sensors from my understanding.

File: pcdsdevices/mirror.py [hunk 1]
```diff
@@ -562,6 +562,11 @@ class KBOMirror(BaseInterface, GroupDevice):
     us_rtd = Cpt(EpicsSignalRO, ':RTD:BEND:US:1_RBV', kind='normal')
     ds_rtd = Cpt(EpicsSignalRO, ':RTD:BEND:DS:1_RBV', kind='normal')
 
+    # Cooling water flow and pressure sensors
+    cool_flow1  = Cpt(EpicsSignalRO, ':FLOW:1_RBV', kind='normal')
+    cool_flow2  = Cpt(EpicsSignalRO, ':FLOW:2_RBV', kind='normal')
```
Comment (ZryletTC) [inline pcdsdevices/mirror.py:567]:
```suggestion
    cool_flow1 = Cpt(EpicsSignalRO, ':FLOW:1_RBV', kind='normal')
    cool_flow2 = Cpt(EpicsSignalRO, ':FLOW:2_RBV', kind='normal')
```

**Comment by nrwslac** [comment]
> > One concern is kbo mirrors using this class that do not have sensors installed will show PV discon here, but that seems okay. All KBOs will receive these sensors from my understanding.
> 
> One way to alleviate this concern is to implement the additional sensors in a child class, and apply the new class in happi when the sensor PVs are added:
> 
> ```
> class CoolKBOMirror(KBOMirror):
>     # Cooling water flow and pressure sensors
>     cool_flow1  = Cpt(EpicsSignalRO, ':FLOW:1_RBV', kind='normal')
>     cool_flow2  = Cpt(EpicsSignalRO, ':FLOW:2_RBV', kind='normal')
>     cool_press = Cpt(EpicsSignalRO, ':PRESS:1_RBV', kind='normal')
> ```
> 
> (with a more normal class name, probably)

Personally I don't like using more inheritance than necessary; I prefer this method. But I am open to doing it in child class.

**Review by ZryletTC — CHANGES_REQUESTED** [review]
Looks good but yeah, these components should be added in a child class if some devices will have them and some will not.

**Comment by ZLLentz** [comment]
I think it's correct to be inheritance-adverse. The thing to be wary of is complicated, deep inheritance trees. A small class that slightly extends a neighboring class in the same file isn't much of a concern I think.

The main hazards related to disconnected PVs in a screen are:
- Users seeing a disconnected PV and thinking something is broken, creating an extra support call
- There's a ... bug? in the positioners where a disconnected positioner PV is pretty fatal when requesting a move, but that doesn't apply here.

File: pcdsdevices/mirror.py (hunk 1 above)
Comment (nrwslac) [reply pcdsdevices/mirror.py]: fixed

**Comment by nrwslac** [comment]
> I think it's correct to be inheritance-adverse. The thing to be wary of is complicated, deep inheritance trees. A small class that slightly extends a neighboring class in the same file isn't much of a concern I think.
> 
> The main hazards related to disconnected PVs in a screen are:
> 
> * Users seeing a disconnected PV and thinking something is broken, creating an extra support call
> * There's a ... bug? in the positioners where a disconnected positioner PV is pretty fatal when requesting a move, but that doesn't apply here.

Sounds like the benefits are outweighing the costs here. I updated to a child class.

File: pcdsdevices/mirror.py [hunk 2]
```diff
@@ -681,6 +676,24 @@ def format_status_info(self, status_info):
...
+    prefix : str
+        Base PV for the mirror.
+
+    name : str
+        Alias for the device.
+    """
+    # Cooling water flow and pressure sensors
+    cool_flow1 = Cpt(EpicsSignalRO, ':FLOW:1_RBV', kind='normal')
+    cool_flow2 = Cpt(EpicsSignalRO, ':FLOW:2_RBV', kind='normal')
+    cool_press = Cpt(EpicsSignalRO, ':PRESS:1_RBV', kind='normal')
```
Comment (ZryletTC) [inline pcdsdevices/mirror.py:697]: Needs two blank lines between classes.
```suggestion


class KBOMirrorHE(KBOMirror):
    """
    Kirkpatrick-Baez Mirror with Bender Axes and Cooling.

    1st gen Toyama designs with LCLS-II Beckhoff motion architecture.

    Parameters
    ----------
    prefix : str
        Base PV for the mirror.

    name : str
        Alias for the device.
    """
    # Cooling water flow and pressure sensors
    cool_flow1 = Cpt(EpicsSignalRO, ':FLOW:1_RBV', kind='normal')
    cool_flow2 = Cpt(EpicsSignalRO, ':FLOW:2_RBV', kind='normal')
    cool_press = Cpt(EpicsSignalRO, ':PRESS:1_RBV', kind='normal')


```

**Review by ZryletTC — APPROVED** [review]
Looks good 👍

File: pcdsdevices/mirror.py (hunk 2 above)
Comment (nrwslac) [reply pcdsdevices/mirror.py]: fixed

**Review by ZLLentz — CHANGES_REQUESTED** [review]
Just needs the pre-release documentation as per the checklist.
```
cd docs
./pre-release-notes.sh 949 kbo-fixes
```

File: docs/source/upcoming_release_notes/949-kbo-fixes.rst [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Device Updates
+--------------
+Flow/Pressure sensor hardware added to MR2K4 KBO. 
+Update software for readbacks.                                        
+
+Added KBOMirrorHE class
+Added Y axis to KBOMirror status
+New Devices
+-----------
+- N/A
```
Comment (ZryletTC) [inline docs/source/upcoming_release_notes/949-kbo-fixes.rst:21]:
```suggestion
- Added Y axis to KBOMirror status.

New Devices
-----------
- Added KBOMirrorHE class to be used with KBOMirrors with cooling, like MR2K4 KBO.
```

File: docs/source/upcoming_release_notes/949-kbo-fixes.rst [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+--------
+- N/A
+
+Maintenance
+-----------
+- N/A
+
+Contributors
+------------
+Nick Waters
```
Comment (ZryletTC) [inline docs/source/upcoming_release_notes/949-kbo-fixes.rst:33]:
```suggestion
- nrwslac
```

Comment (nrwslac) [reply docs/source/upcoming_release_notes/949-kbo-fixes.rst]: fixed

**Review by ZLLentz — APPROVED** [review]
Let's get it merged, lookin good!

**Pull request merged**